src/editors/playgroundController.ts

```diff
@@ -492,33 +492,33 @@ export default class PlaygroundController {
     }
 
     try {
-      const progressResult = await vscode.window.withProgress(
+      return await vscode.window.withProgress(
         {
           location: ProgressLocation.Notification,
           title: 'Running MongoDB playground...',
           cancellable: true,
         },
-        async (progress, token) => {
-          token.onCancellationRequested(() => {
-            // If a user clicked the cancel button terminate all playground scripts.
-            this._languageServerController.cancelAll();
-
-            return { result: undefined };
+        async (progress, token): Promise<ShellEvaluateResult> => {
+          // eslint-disable-next-line @typescript-eslint/no-misused-promises
+          return new Promise(async (resolve, reject) => {
+            token.onCancellationRequested(() => {
+              // If a user clicked the cancel button terminate all playground scripts.
+              this._languageServerController.cancelAll();
```

Comment:
🤔 Also unrelated to these changes, so no changes needed, but I feel I should ask, can't the language server only be running one script at a time?

Reply:
I think that cancelAll relates here to all sent requests that belong to one language server controller.